### PR TITLE
Remove FastSim dependecy in Validation/RecoParticleFlow/plugins/PFAnalysis.cc

### PR DESCRIPTION
**Pull request by silviodonato**
#### PR description:

This PR removes the FastSim dependency in Validation/RecoParticleFlow/plugins/PFAnalysis.cc by replacing `Transform3DPJ` with `Transform3D` from root. This PR fixes the error
```
  ****ERROR: Dependency violation (direct): Validation/RecoParticleFlow/plugins/PFAnalysis.cc FastSimulation/CaloGeometryTools/interface/Transform3DPJ.h
  *Validation/RecoParticleFlow/plugins/PFAnalysis.cc (Validation/RecoParticleFlow/plugins/PFAnalysis.cc)
    *FastSimulation/CaloGeometryTools/interface/Transform3DPJ.h (FastSimulation/CaloGeometryTools/interface/Transform3DPJ.h)
```
See https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc820/CMSSW_11_2_X_2020-11-23-2300/depViolationLogs/Validation/RecoParticleFlow

#### PR validation:

I tested `ReleaseDepsChecks.pl`and the errors has disappeared after the fix .

### CODE REVIEW

**Comment by silviodonato** [comment]
please test

File: Validation/RecoParticleFlow/plugins/BuildFile.xml [hunk 1]
```diff
@@ -14,6 +14,7 @@
   <use name="CommonTools/UtilAlgos"/>
   <use name="Geometry/HcalTowerAlgo"/>
   <use name="MagneticField/Records"/>
+  <use name="FastSimulation/CaloGeometryTools"/>
```
Comment (davidlange6) [inline Validation/RecoParticleFlow/plugins/BuildFile.xml:17]: this doesn't seem like a dependency we want? In any case it looks like Transform3DPJ.h is just copied from root - perhaps the better solution is to depend on root directly;

https://github.com/cms-sw/cmssw/issues/26016

**Comment by silviodonato** [comment]
please abort

**Comment by silviodonato** [comment]
please test

File: Validation/RecoParticleFlow/plugins/BuildFile.xml (hunk 1 above)
Comment (silviodonato) [reply Validation/RecoParticleFlow/plugins/BuildFile.xml]: Thanks David, I've implemented your suggestion

**Comment by jfernan2** [comment]
HI @silviodonato  I am puzzled by the huge number of changes produced on wf 11642.911 (not a fastsim sample), any clue?
https://tinyurl.com/y2x262ok

**Comment by silviodonato** [comment]
please test
@jfernan2 that is the DD4HEP workflows, that as some problems. Let's try to run it one more time ( see https://github.com/cms-sw/cmssw/issues/32249 )

**Comment by jfernan2** [comment]
+1
Thanks @silviodonato  !

**Comment by silviodonato** [comment]
+1

**Pull request merged**